internal/test/writer.go

```diff
@@ -0,0 +1,27 @@
+package test
+
+import (
+	"io"
+)
+
+// WriterWithHook is an io.Writer that calls a hook function after every write.
+// This is useful for testing.
```

Comment:
This description is a bit redundant. Why is it useful?

Reply:
I'll update it